compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -506,6 +506,20 @@ static LogicalResult setFftOpConfig(spirv::ResourceLimitsAttr limits,
 // Reduction Default Configuration
 //===----------------------------------------------------------------------===//
 
+// Check if the given function contains an op that may require a broadcast of
+// the reduced result.
+static bool isFusedWithBroadcast(linalg::LinalgOp reduce) {
+  func::FuncOp entryPoint = reduce->getParentOfType<func::FuncOp>();
+  int64_t reducedRank =
+      reduce->getResult(0).getType().cast<ShapedType>().getRank();
+  bool hasBroadcast = false;
+  entryPoint.walk([&](linalg::LinalgOp linalgOp) {
+    if (reduce != linalgOp && linalgOp.getNumLoops() > reducedRank)
+      hasBroadcast = true;
```

Comment:
Nit: use `return WalkResult::interrupt` to abort and check that for function return?